src/benchmarks/micro/corefx/System.Collections/Sort.cs

```diff
@@ -11,34 +11,48 @@
 namespace System.Collections
 {
     [BenchmarkCategory(Categories.CoreCLR, Categories.Collections, Categories.GenericCollections)]
-    [GenericTypeArguments(typeof(int))] // value type
-    [GenericTypeArguments(typeof(string))] // reference type
+    [GenericTypeArguments(typeof(int))] // value type, Array sort in native code
+    [GenericTypeArguments(typeof(IntStruct))] // custom value type, sort in managed code
+    [GenericTypeArguments(typeof(IntClass))] // custom reference type, sort in managed code, compare fast
+    [GenericTypeArguments(typeof(BigStruct))] // custom value type, sort in managed code
+    [GenericTypeArguments(typeof(string))] // reference type, compare slow
     [InvocationCount(InvocationsPerIteration)]
-    public class Sort<T>
+    public class Sort<T> where T : IComparable<T>
     {
+        static readonly ComparableComparerClass _comparableComparerClass = new ComparableComparerClass();
         private const int InvocationsPerIteration = 1000;
```

Comment:
@billwert I have kept the `1000` InvocationsPerIteration, but do note this is very small for the small types.